binance_margin.go

```diff
@@ -0,0 +1,258 @@
+package binance
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+)
+
+const (
+	listenKeyTypeMarginAPI         = "sapi/v1/userDataStream"
+	listenKeyTypeIsolatedMarginAPI = "sapi/v1/userDataStream/isolated"
+)
+
+// CreateListenKeyMargin create a listen key for user data stream
+func (bc *Client) CreateListenKeyMargin() (string, error) {
+	return bc.createListenKey(listenKeyTypeMarginAPI)
+}
+
+// KeepListenKeyAliveMargin keep it alive
+func (bc *Client) KeepListenKeyAliveMargin(listenKey string) error {
+	return bc.keepListenKeyAlive(listenKey, listenKeyTypeMarginAPI)
+}
+
+// CreateListenKeyIsolatedMargin create a listen key for user data stream
+func (bc *Client) CreateListenKeyIsolatedMargin() (string, error) {
+	return bc.createListenKey(listenKeyTypeIsolatedMarginAPI)
+}
+
+// KeepListenKeyAliveIsolatedMargin keep it alive
+func (bc *Client) KeepListenKeyAliveIsolatedMargin(listenKey string) error {
+	return bc.keepListenKeyAlive(listenKey, listenKeyTypeIsolatedMarginAPI)
+}
+
+type marginCommonResult struct {
+	TranID uint64 `json:"tranId"`
+}
+
+func (bc *Client) CrossMarginTransfer(asset, amount string, spotToMargin bool) (uint64, *FwdData, error) {
+	transType := ""
+	if spotToMargin {
+		transType = "1"
```

Comment:
this should be set as constant

Reply:
this param is only being used by this API and determined by param ```spotToMargin```, in isolated margin Binance uses another type for this. So I don't think we need to define the constant values for this